nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -819,7 +819,8 @@ in {
         ++ (optional (versionOlder cfg.package.version "25") (upgradeWarning 24 "22.11"))
         ++ (optional (versionOlder cfg.package.version "26") (upgradeWarning 25 "23.05"))
         ++ (optional (versionOlder cfg.package.version "27") (upgradeWarning 26 "23.11"))
-        ++ (optional (versionOlder cfg.package.version "28") (upgradeWarning 27 "24.05"));
+        ++ (optional (versionOlder cfg.package.version "28") (upgradeWarning 27 "24.05"))
+        ++ (optional (versionOlder cfg.package.version "29") (upgradeWarning 28 "24.11"));
```

Comment:
As i am currently trying to upgrade a nextcloud server from 23.11 to 24.05 that warning feels a bit off as the defaults and the generated warning messages to not fit.

On `stateVersion` 24.05 we set a nextcloud29, but do not warn for having nextcloud29 available in 24.05 as default.

Also i have the feeling that the generated warning is not totally correct as its already including 24.11 information.


```
warning: Git tree '/home/shawn/dev/nix-configuration' is dirty
trace: warning: A legacy Nextcloud install (from before NixOS 24.11) may be installed.

After nextcloud28 is installed successfully, you can safely upgrade
to 29. The latest version available is Nextcloud29.

Please note that Nextcloud doesn't support upgrades across multiple major versions
(i.e. an upgrade from 16 is possible to 17, but not 16 to 18).

The package can be upgraded by explicitly declaring the service-option
`services.nextcloud.package`.
```

Be aware of the 24.11 in the warning.

Is that intended or a bug?